pkg/controller/pravegacluster/upgrade_test.go

```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
+			var (
+				err, err1, err2, err3 error
+				foundPravega          *v1beta1.PravegaCluster
+				client                client.Client
+			)
+			BeforeEach(func() {
+				client = fake.NewFakeClient(p)
+				r = &ReconcilePravegaCluster{client: client, scheme: s}
+				_, _ = r.Reconcile(req)
+				foundPravega = &v1beta1.PravegaCluster{}
+				_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+				_, err1 = r.syncControllerVersion(foundPravega)
+				deploy := pravega.MakeControllerDeployment(foundPravega)
+				r.client.Create(context.TODO(), deploy)
+				_, err2 = r.syncControllerVersion(foundPravega)
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: deploy.Name, Namespace: foundPravega.Namespace}, deploy)
+				deploy.Status.UpdatedReplicas = 5
+				deploy.Status.Replicas = 3
+				r.client.Update(context.TODO(), deploy)
+				foundPravega.Status.TargetVersion = "0.5.0"
+				r.client.Update(context.TODO(), foundPravega)
+				_, err = r.syncControllerVersion(foundPravega)
+				condition := appsv1.DeploymentCondition{
+					Type:    "Progressing",
+					Status:  corev1.ConditionFalse,
+					Reason:  "ProgressDeadlineExceeded",
+					Message: "",
+				}
+				deploy.Status.Conditions = append(deploy.Status.Conditions, condition)
+				r.client.Update(context.TODO(), deploy)
+				_, err3 = r.syncControllerVersion(foundPravega)
+			})
+			It("Error should be nil", func() {
+				Ω(err).Should(BeNil())
+			})
+			It("Error should not be nil", func() {
+				Ω(strings.ContainsAny(err1.Error(), "failed to get deployment ()")).Should(Equal(true))
+			})
+			It("Error should not be nil", func() {
+				Ω(strings.ContainsAny(err2.Error(), "target version is not set")).Should(Equal(true))
+			})
+			It("Error should not be nil", func() {
+				Ω(strings.ContainsAny(err3.Error(), "failed due to ProgressDeadlineExceeded")).Should(Equal(true))
+			})
+		})
+		Context("syncSegmentStoreVersion", func() {
+			var (
+				err, err1    error
+				foundPravega *v1beta1.PravegaCluster
+				client       client.Client
+			)
+			BeforeEach(func() {
+				client = fake.NewFakeClient(p)
+				r = &ReconcilePravegaCluster{client: client, scheme: s}
+				_, _ = r.Reconcile(req)
+				foundPravega = &v1beta1.PravegaCluster{}
+				_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+				sts := pravega.MakeSegmentStoreStatefulSet(foundPravega)
+				r.client.Create(context.TODO(), sts)
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: sts.Name, Namespace: foundPravega.Namespace}, sts)
+				_, err = r.syncSegmentStoreVersion(foundPravega)
+				foundPravega.Status.TargetVersion = "0.5.0"
+				sts.Status.UpdatedReplicas = 5
+				sts.Status.Replicas = 3
+				r.client.Update(context.TODO(), sts)
+				r.client.Update(context.TODO(), foundPravega)
+				_, err1 = r.syncSegmentStoreVersion(foundPravega)
+			})
+			It("Error should be nil", func() {
+				Ω(err).ShouldNot(BeNil())
+			})
+			It("Error should be nil", func() {
```

Comment:
Assertion description is not correct here.

Reply:
fixed